packages/plugin-core/src/commands/VaultAddCommand.ts

```diff
@@ -133,6 +138,83 @@ export class VaultAddCommand extends BasicCommand<CommandOpts, CommandOutput> {
     };
   }
 
+  async gatherVaultSelfContained(
+    sourceType: VaultRemoteSource
+  ): Promise<CommandOpts | undefined> {
+    // If the vault name already exists, creating a vault with the same name would break things
+
+    if (sourceType === VaultType.LOCAL) {
+      // Local vault
+      // For self contained vaults, we'll have the vault name match the folder for
+      // now. We can make this flexible later if that's a better UX, or give
+      // instructions on the wiki on how to change the name later.
+      const vaultName = await VSCodeUtils.showInputBox({
+        title: "Vault name",
+        prompt: "Name for the new vault",
+        placeHolder: "my-vault",
+      });
+      // If empty, then user cancelled the prompt
+      if (PickerUtilsV2.isInputEmpty(vaultName)) return;
+
+      return {
+        type: sourceType,
+        name: vaultName,
+        path: path.join(
+          FOLDERS.DEPENDENCIES,
+          FOLDERS.LOCAL_DEPENDENCY,
+          vaultName
+        ),
+        isSelfContained: true,
+      };
+    } else {
+      // Remote vault
+      const remote = await VSCodeUtils.showInputBox({
+        title: "Remote URL",
+        prompt: "Enter the URL for the git remote",
+        placeHolder: "git@github.com:dendronhq/dendron.git",
+        ignoreFocusOut: true,
+      });
+      // Cancelled
+      if (PickerUtilsV2.isInputEmpty(remote)) return;
+
+      // Calculate the vault name from the remote. If that fails, ask the user for a unique name to use.
+      const vaultName: string | undefined = GitUtils.getRepoNameFromURL(remote);
+
+      return {
+        type: sourceType,
+        name: vaultName,
+        path: path.join(
+          FOLDERS.DEPENDENCIES,
+          GitUtils.remoteUrlToDependencyPath({
+            vaultName,
+            url: remote,
+          })
+        ),
+        pathRemote: remote,
+        isSelfContained: true,
+      };
+    }
+  }
+
+  async gatherInputs(): Promise<CommandOpts | undefined> {
+    const sourceTypeSelected = await VSCodeUtils.showQuickPick([
+      { label: VaultType.LOCAL, picked: true },
+      { label: VaultType.REMOTE },
+    ]);
+    if (!sourceTypeSelected) {
+      return;
+    }
+    const sourceType = sourceTypeSelected.label;
+
+    const { config } = ExtensionProvider.getDWorkspace();
+    if (config.dev?.enableSelfContainedVaults) {
+      return this.gatherVaultSelfContained(sourceType);
+    } else {
+      // A "standard", non self contained vault
```

Comment:
super nit: maybe move this comment (or copy it) to the method header of `gatherVaultStandard` - just makes it a little clearer while reading it what 'standard' refers to.